src/nvim/api/vim.c

```diff
@@ -951,12 +951,18 @@ Buffer nvim_create_buf(Boolean listed, Boolean scratch, Error *err)
 /// @return Channel id, or 0 on error
 Integer nvim_open_term(Buffer buffer, DictionaryOf(LuaRef) opts, Error *err)
   FUNC_API_SINCE(7)
+  FUNC_API_CHECK_TEXTLOCK_ALLOW_CMDWIN
 {
   buf_T *buf = find_buffer_by_handle(buffer, err);
   if (!buf) {
     return 0;
   }
 
+  if (cmdwin_type != 0 && buf == curbuf) {
+    api_set_error(err, kErrorTypeException, "%s", _(e_cmdwin));
```

Comment:
Does this disallow `nvim_open_term` in cmdwin? If so, why is it marked as FUNC_API_CHECK_TEXTLOCK_ALLOW_CMDWIN ?

Reply:
It's allowed in the cmdwin, as long as you don't try to open a terminal in the cmdwin's buffer.